src/app/leverage/provider.tsx

```diff
@@ -144,6 +144,7 @@ export function LeverageProvider(props: { children: any }) {
   const { balances, forceRefetchBalances } = useBalances(
     address,
     leverageTokenAddresses,
+    chainId !== ARBITRUM.chainId,
```

Comment:
Avoid calling `balanceOf` if the chain does not match the tokens that will be passed to it.

Reply:
`disabled` is a good addition but what about just passing an empty tokens array for non Arbitrum tokens at the moment? Soon we would need to return an array for Ethereum too here.

Reply:
@janndriessen good call, seems like `undefined` will work too, tweaked the implementation to better support other chains